### DISCUSSION TITLE
Get index of array element

### DISCUSSION THREAD

**Discussion by beescuit** [github/codeql#14999] [Q&A] [answered]
In javascript, would it be possible to retrieve the index of an element of a `ArrayExpr`, retrieved via `getAnElement`, relative to the array?

**Comment by smowton** [comment]
Try using `getElement(int i)` instead of `getAnElement`.

**Reply by beescuit** [reply]
What if I don't know the index? e.g.

```javascript
const a = [
  {name: "a"},
  {name: "b"},
  {name: "c"}
]
```

And I want to query the index of the element that has the name "b"?

**Reply by smowton** [reply, marked as answer]
```ql
from int i, Expr bElement, ArrayExpr array
where array.getElement(i) = bElement and
/* check bElement has desired 'name: b' member */
select array, bElement, i
```

**Reply by beescuit** [reply]
Oh, that feels obvious in hindsight lol. Thank you!
